code/Common/src/main/java/net/selenate/common/comms/req/SeReqWindowFrameSwitch.java

```diff
@@ -1,61 +1,98 @@
 package net.selenate.common.comms.req;
 
-import java.util.List;
-import net.selenate.common.exceptions.SeNullArgumentException;
+import java.util.*;
+
 import net.selenate.common.SelenateUtils;
+import net.selenate.common.comms.*;
 
 public final class SeReqWindowFrameSwitch implements SeCommsReq {
   private static final long serialVersionUID = 45749879L;
 
-  private final List<Integer> framePath;
+  public static enum FrameSwitchMethod {
+    PATH,
+    NAME,
+    SELECTOR
+  }
+
+  private final FrameSwitchMethod method;
+  private final List<Integer>     framePath;
+  private final String            frameName;
+  private final SeElementSelector frameSelector;
 
   public SeReqWindowFrameSwitch(final List<Integer> framePath) {
-    this.framePath = framePath;
-    validate();
+    this.method        = FrameSwitchMethod.PATH;
+    this.framePath     = SelenateUtils.guardNull(framePath, "FramePath");
+    this.frameName     = null;
+    this.frameSelector = null;
+  }
+
+  public SeReqWindowFrameSwitch(final String frameName) {
```

Comment:
I'm not a fan of multiple constructors in `Req`/`Res` classes.
Shouldn't the client interface provide them?